### PR TITLE
Windows test fixes.

### PR DESCRIPTION
**Pull request by pq**
Baby steps towards #113.

### CODE REVIEW

**Comment by pq** [comment]
@keertip PTAL?

**Comment by keertip** [comment]
lgtm

File: lib/src/util.dart [hunk 1]
```diff
@@ -21,16 +21,17 @@ final _pubspec = new RegExp(r'^[_]?pubspec\.yaml$');
 /// current [packageName].
 String createLibraryNamePrefix(
     {String libraryPath, String projectRoot, String packageName}) {
-  var libraryDirectory = p.dirname(libraryPath);
-  var path = p.relative(libraryDirectory, from: projectRoot);
-  // Drop 'lib/'
+  // Use the posix context to canonicalize separators (`\`).
+  var libraryDirectory = p.posix.dirname(libraryPath);
```
Comment (donny-dont) [inline lib/src/util.dart]: I don't think you want to start using p.posix until you do a join.

File: lib/src/util.dart [hunk 2]
```diff
@@ -21,16 +21,17 @@ final _pubspec = new RegExp(r'^[_]?pubspec\.yaml$');
...
+  var path = p.posix.relative(libraryDirectory, from: projectRoot);
+  // Drop 'lib/'.
   var segments = p.split(path);
   if (segments[0] == 'lib') {
-    path = p.joinAll(segments.sublist(1));
+    path = p.posix.joinAll(segments.sublist(1));
   }
-  // Replace separators
+  // Replace separators.
   path = path.replaceAll('/', '.');
```
Comment (donny-dont) [inline lib/src/util.dart]: If it didn't hit the if statement above the paths aren't normalized to posix style.

**Comment by donny-dont** [comment]
Sorry for being late to the party. I did some fixes like this locally and I believe what you want to do is use the os specific path until you do a join. At that point for sure everything is in `/path/to/foo` form.

**Comment by pq** [comment]
Nothing to be sorry about!  Thanks for the catch.  I've not had much time on a windows box lately so this input is invaluable.  Really!

If you feel up for it, maybe through together a Pull Request?  If not, I can throw some changes your way to review but I won't have access to my windows machine until early next week...

Cheers!

**Comment by donny-dont** [comment]
I didn't end up doing a patch because I wasn't sure at the time on how to properly deal with windows/posix differences.

I found that with the analyzer if I used URIs that the problem was simplified as those are all in a posix compatible form. LibraryElement.definingCompilationUnit.source.uri is what I use when I get the library. From there I wrote a wrapper over path that handled things.

Applicable code at
https://github.com/dogma-dart/dogma-codegen/blob/master/lib/path.dart

Now I'm pretty happy with it for my own codegen library and I would probably suggest something similar for the linter. Not sure how that might change things but for me it all fell into place.

Happy to discuss more about it and if you want it done a certain way I can potentially hack on that.

**Pull request merged**